tests/integration/agents/test_gatekeeper_feedback.py

```diff
@@ -0,0 +1,271 @@
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'CustomResourceDefinition', 'resource_api_version': 'apiextensions.k8s.io/v1', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(content="The YAML syntax is incorrect in your example."),
+            ],
+            True,
+        ),
+        (
+            "user requests alternative approach",
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'HorizontalPodAutoscaler', 'resource_api_version': 'autoscaling/v2', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(
+                    content="Is there a different way to achieve this without using HPA?"
+                ),
+            ],
+            False,
+        ),
+        # Ambiguous feedback
+        (
+            "user gives minimal acknowledgment",
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'ServiceAccount', 'resource_api_version': 'v1', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(content="Hmm, okay."),
+            ],
+            False,
+        ),
+        (
+            "user gives brief confirmation",
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'Role', 'resource_api_version': 'rbac.authorization.k8s.io/v1', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(content="I see."),
+            ],
+            False,
+        ),
+        # can you clarify more?
+        (
+            "user asks for clarification",
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'ConfigMap', 'resource_api_version': 'v1', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(content="Can you clarify more?"),
+            ],
+            False,
+        ),
+        (
+            "user asks for clarification",
+            [
+                SystemMessage(
+                    content="The user query is related to: "
+                    "{'resource_kind': 'ConfigMap', 'resource_api_version': 'v1', 'resource_name': '', 'namespace': ''}"
+                ),
+                HumanMessage(content="Can you elaborate more?"),
+            ],
+            False,
+        ),
+        # give shorter explanation
+        (
+            "user gives shorter explanation",
+            [
+                SystemMessage(content="The user query is related to: "),
+                HumanMessage(content="give me a shorter explanation"),
+            ],
+            False,
+        ),
+    ],
+)
```

Comment:
Can you add a test case where there is little more conversation history?

Reply:
currently, feedback is only considering the user query. It doesn't consider conversation history. 
That's why I thought, it doesn't bring anything as it only uses only latest human message. 
There is another reason, where nano model cannot handle too much data why I only consider only user query.